src/server/models/listItem/listItem.ts

```diff
@@ -103,6 +111,41 @@ export async function findListItemById(id: string | number) {
   }
 }
 
+export async function findListItems(options: {
+    listIds?: number[],
+    listItemIds?: number[],
+    statuses?: Status[],
+    isAnnualReview?: boolean
+}) {
+  try {
+    const { listIds, listItemIds, statuses, isAnnualReview } = options;
+    if ((!listIds || !listIds.length) && (!listItemIds || !listItemIds?.length)) {
+      const message = "List ids or list item ids must be specified to find list items";
+      logger.error(message);
+      return { error: Error(message) };
+    }
+    const result = await prisma.listItem.findMany({
+      where: {
+        ...(listIds != null && { listId: { in: listIds }}),
```

Comment:
this suggests that listIds can't be optional, and can't be empty arrays (although empty array is not a big deal, undefined/null is a bigger issue for ts/js). 

if you've handled a nullish value with a guard clause, typically tsc will be able to unwrap your values further down (i.e. no need for null check), but it isn't able to here since the if/guard statement is too complicated. 

the only two optional options are statuses and annualReview. so is it worth changing the type so that listId and listItemIds are required? 

The benefit of doing something like 
```.ts
...(listItemIds != null && { id: { in: listItemIds }}),
```
allows you to do something like "if listItemIds is defined, return `{ id: { in: listItemIds }}`. 

listItemIds must always be defined (due to guard clause), so you don't need to do `...(listItemIds != null`

Reply:
the purpose of this validation is to ensure either listIds or listItemIds are populated so we're not left with an empty where clause and all records are returned.  Therefore, if I understand your point, I think `...(listItemIds != null && { id: { in: listItemIds }})` is still needed as you can't guarantee listItemIds will indeed be populated.